When trying to run py-flake8@3.8.2, I get this (py-flake8@3.7.8 works):
```
Traceback (most recent call last):
  File ".../opt/spack/linux-ubuntu18.04-x86_64/gcc-9.3.0/py-flake8-3.8.2-btdipsrxs33hysydumnm4sgshjf7ldd3/bin/flake8", line 6, in <module>
    from pkg_resources import load_entry_point
  File ".../opt/spack/linux-ubuntu18.04-x86_64/gcc-9.3.0/py-setuptools-47.1.1-6ttmhdtese4dk3wvvdotw7abcd3lgdv2/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3261, in <module>
    @_call_aside
  File ".../opt/spack/linux-ubuntu18.04-x86_64/gcc-9.3.0/py-setuptools-47.1.1-6ttmhdtese4dk3wvvdotw7abcd3lgdv2/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3245, in _call_aside
    f(*args, **kwargs)
  File ".../opt/spack/linux-ubuntu18.04-x86_64/gcc-9.3.0/py-setuptools-47.1.1-6ttmhdtese4dk3wvvdotw7abcd3lgdv2/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3274, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File ".../opt/spack/linux-ubuntu18.04-x86_64/gcc-9.3.0/py-setuptools-47.1.1-6ttmhdtese4dk3wvvdotw7abcd3lgdv2/lib/python3.7/site-packages/pkg_resources/__init__.py", line 584, in _build_master
    ws.require(__requires__)
  File ".../opt/spack/linux-ubuntu18.04-x86_64/gcc-9.3.0/py-setuptools-47.1.1-6ttmhdtese4dk3wvvdotw7abcd3lgdv2/lib/python3.7/site-packages/pkg_resources/__init__.py", line 901, in require
    needed = self.resolve(parse_requirements(requirements))
  File ".../opt/spack/linux-ubuntu18.04-x86_64/gcc-9.3.0/py-setuptools-47.1.1-6ttmhdtese4dk3wvvdotw7abcd3lgdv2/lib/python3.7/site-packages/pkg_resources/__init__.py", line 787, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'importlib-metadata' distribution was not found and is required by flake8
```

--- a/var/spack/repos/builtin/packages/py-flake8/package.py
+++ b/var/spack/repos/builtin/packages/py-flake8/package.py
@@ -59,10 +59,11 @@ class PyFlake8(PythonPackage):
     depends_on('py-mccabe@0.2.1:0.4',                 when='@2.5.0:2.5.999', type=('build', 'run'))
 
     # Python version-specific backports
-    depends_on('py-enum34',       when='@3.0.0: ^python@:3.3', type=('build', 'run'))
-    depends_on('py-typing',       when='@3.7.0: ^python@:3.4', type=('build', 'run'))
-    depends_on('py-configparser', when='@3.0.0: ^python@:3.1', type=('build', 'run'))
-    depends_on('py-functools32',  when='@3.7.4: ^python@:3.1', type=('build', 'run'))
+    depends_on('py-importlib-metadata', when='@3.8.0: ^python@:3.7', type=('build', 'run'))
+    depends_on('py-enum34',             when='@3.0.0: ^python@:3.3', type=('build', 'run'))
+    depends_on('py-typing',             when='@3.7.0: ^python@:3.4', type=('build', 'run'))
+    depends_on('py-configparser',       when='@3.0.0: ^python@:3.1', type=('build', 'run'))
+    depends_on('py-functools32',        when='@3.7.4: ^python@:3.1', type=('build', 'run'))
 
     def patch(self):
         """Filter pytest-runner requirement out of setup.py."""

